### PR TITLE
refactor: create trackEvent function and apply updates in relevant files

### PR DESCRIPTION
**Pull request by Mike-Heneghan**
## What

- As per: https://github.com/theopensystemslab/planx-new/pull/2597#discussion_r1448669108
- Consolidating individual tracking event functions into a single function with a case statement

## Why

- Step towards clearer organisation and documentation of analyticsProvider
- Should hopefully be a clearer pattern for adding tracking of new events

### CODE REVIEW

**Comment by Mike-Heneghan** [comment]
[Improve readability / documentation of analyticsProvider.tsx](https://trello.com/c/zVTLbjUn/2882-improve-readability-documentation-of-analyticsprovidertsx)

File: editor.planx.uk/src/pages/FlowEditor/lib/analyticsProvider.tsx [hunk 1]
```diff
@@ -85,32 +85,84 @@ type Metadata =
...
     direction?: AnalyticsLogDirection,
     analyticsSessionId?: number,
   ) => Promise<void>;
 }>({
   createAnalytics: () => Promise.resolve(),
-  trackHelpClick: () => Promise.resolve(),
-  trackNextStepsLinkClick: () => Promise.resolve(),
-  trackFlowDirectionChange: () => Promise.resolve(),
-  trackBackwardsNavigation: () => Promise.resolve(),
+  trackEvent: () => Promise.resolve(),
```
Comment (DafyddLlyr) [inline editor.planx.uk/src/pages/FlowEditor/lib/analyticsProvider.tsx:164]: Much neater and cleaner 💪

**Review by DafyddLlyr — APPROVED** [review]
Really strong refactor - this seems like it would be much easier to maintain and expand.

A few typo/type suggestions but no showstoppers.

File: editor.planx.uk/src/pages/FlowEditor/lib/analyticsProvider.tsx [hunk 2]
```diff
@@ -289,61 +337,65 @@ export const AnalyticsProvider: React.FC<{ children: React.ReactNode }> = ({
     return !shouldTrackAnalytics || !lastVisibleNodeAnalyticsLogId;
   }
 
-  async function trackHelpClick(metadata?: HelpClickMetadata) {
-    if (shouldSkipTracking()) return;
-
+  async function updateMetadata(mutation: any, metadata: Metadata) {
```
[the remark this answers is not in the document]
Comment (Mike-Heneghan) [reply editor.planx.uk/src/pages/FlowEditor/lib/analyticsProvider.tsx]: Thanks for spotting this @DafyddLlyr, I did the classic "I'll come back to this later and totally forgot 😅" should be fixed here: a7ac3b157369b1fec618ecbfd85927dc4615725c

File: editor.planx.uk/src/pages/FlowEditor/lib/analyticsProvider.tsx [hunk 3]
```diff
@@ -85,32 +85,84 @@ type Metadata =
...
+ */
+type BackwardsNavigation = {
+  event: "backwardsNavigation";
+  metadata: null;
+  initiator: BackwardsNavigationInitiatorType;
+  nodeId?: string;
+};
+
+/**
+ * The flow direction when a new analytics session is create is
```
[the remark this answers is not in the document]
Comment (Mike-Heneghan) [reply editor.planx.uk/src/pages/FlowEditor/lib/analyticsProvider.tsx]: Should be fixed here: f811fe2887f6f30ee7d62ac5f892a94fcda00183

**Pull request merged**